src/UnitTests/AttributeBasedMaps.cs

```diff
@@ -953,5 +953,58 @@ public void Should_validate_successfully()
                 typeof(AutoMapperConfigurationException).ShouldNotBeThrownBy(() => AssertConfigurationIsValid<SecondSource, Dest>());
             }
         }
+
+        public class When_specifying_as_proxy_via_attribute : NonValidatingSpecBase
+        {
+            public class Source
+            {
+                public int Value { get; set; }
+            }
+
+            [AutoMap(typeof(Source), AsProxy = true)]
+            public interface IDest
+            {
+                int Value { get; set; }
+            }
+
+            protected override MapperConfiguration CreateConfiguration() => new(cfg =>
+            {
+                cfg.AddMaps(typeof(When_specifying_as_proxy_via_attribute));
+            });
+
+            [Fact]
+            public void Should_convert_to_interface()
+            {
+                var source = new Source { Value = 15 };
+                IDest dest = Mapper.Map<IDest>(source);
+                dest.Value.ShouldBe(15);
+            }
+        }
+
+        public class When_not_specifying_as_proxy_via_attribute : NonValidatingSpecBase
+        {
+            public class Source
+            {
+                public int Value { get; set; }
+            }
+
+            [AutoMap(typeof(Source), AsProxy = false)]
+            public interface IDest
+            {
+                int Value { get; set; }
+            }
+
+            protected override MapperConfiguration CreateConfiguration() => new(cfg =>
+            {
+                cfg.AddMaps(typeof(When_not_specifying_as_proxy_via_attribute));
+            });
+
+            [Fact]
+            public void Should_not_convert_to_interface()
+            {
+                var source = new Source { Value = 15 };
+                typeof(AutoMapperMappingException).ShouldBeThrownBy(() => Mapper.Map<IDest>(source));
```

Comment:
If we have this  test, we should check smth about the exception, so we know why it failed.

Reply:
Please review the updates